### ISSUE TITLE
Subtitles have the wrong decoding

### ISSUE THREAD

**Issue by duramato** [kitsuneDEV/SubsCleaner.bundle#2]
The subtitles after being scrubbed of the _annoyances_ are left like this:

```
1
00:01:14,880 --> 00:01:16,520
Obrigado. estamos de volta!

2
00:01:17,880 --> 00:01:18,920
N�s voltamos.

3
00:01:20,920 --> 00:01:22,480
Muito obrigado, pessoal.

4
00:01:22,560 --> 00:01:23,800
E sejam bem-vindos.

5
00:01:23,880 --> 00:01:26,680
Bem-vindos � nossa nova casa permanente,

6
00:01:26,920 --> 00:01:30,320
Bem aqui no cora��o verde da inglaterra.

7
00:01:31,520 --> 00:01:32,360
� isso a�.

8
00:01:32,840 --> 00:01:33,640
Isso a�.

9
00:01:37,360 --> 00:01:38,880
Se posso dizer,

10
00:01:38,960 --> 00:01:41,720
N�o � tao verde quanto � inconveniente.

11
00:01:42,000 --> 00:01:43,200
Por que � inconveniente?

12
00:01:43,280 --> 00:01:45,640
Bem, jeremy,
Diga �s senhoras e aos senhores

13
00:01:45,720 --> 00:01:49,040
A raz�o de ter montado a tenda aqui.

14
00:01:49,720 --> 00:01:52,440
� exatamente porque este lugar
.
.
.
```

Notice the wierd utf characters. They are all non utf-8 characters. 
My system's locale:

```
$ locale
LANG=pt_PT.UTF-8
LANGUAGE=
LC_CTYPE="pt_PT.UTF-8"
LC_NUMERIC="pt_PT.UTF-8"
LC_TIME="pt_PT.UTF-8"
LC_COLLATE="pt_PT.UTF-8"
LC_MONETARY="pt_PT.UTF-8"
LC_MESSAGES="pt_PT.UTF-8"
LC_PAPER="pt_PT.UTF-8"
LC_NAME="pt_PT.UTF-8"
LC_ADDRESS="pt_PT.UTF-8"
LC_TELEPHONE="pt_PT.UTF-8"
LC_MEASUREMENT="pt_PT.UTF-8"
LC_IDENT
```

**Comment by duramato** [comment]
The issue here is, _I think_ at least, it assumes the file has utf-8 encoding originally which may not always be correct.

https://github.com/kitsuneDEV/SubsCleaner.bundle/blob/master/Contents/Code/__init__.py#L125

**Comment by kitsuneDEV** [comment]
Thanks for the detailed feedback, I have a day off, so I was already going to look into problems with certain srt files. (As mentioned in the other issue it is really a Plex only problem..)

I'll have to add some code to detect encoding and convert the srt when opening it for processing.

Plex requires subtitles to be UTF-8 when saved though, I'm not sure how strict.

I'm also not sure why some files make it endlessly get stuck in the buffering phase. I check those file and they are in UTF-8

Hope I can figure it out today

Also can you mail me this subtitle file? oscar~@~kitsune.work
I wouldn't be able to easily find a srt file on my system with your issue

**Comment by duramato** [comment]
I've sent the subtitle file. 
While it should open the file with it's encoding, it should save it as UTF-8. But i've had no issues with Plex opening the files regardless of their encoding.

**Comment by kitsuneDEV** [comment]
Understood, just need to make sure no characters will be lost after it's done scrubbing the file.

**Comment by kitsuneDEV** [comment]
Pushed a new version out that includes reading the original encoding and saving in that encoding too. I hope that fixes both issues. Time will tell, as I need to come across a file that makes Plex get stuck at the buffering state due to an error with it's accompanying subtitle file.

The preference panel now has an option to force files to be saved as UTF-8 instead.

**Comment by kitsuneDEV** [comment]
**Actually now the subtitle file with other encoding turns into a blank file.. hold off on the 0.94 release!**

**Comment by kitsuneDEV** [comment]
v0.95 appears to be working as expected. Please test carefully.

Hope all is working now, this Plex stuff is draining my life force..

**Comment by duramato** [comment]
Will give it a try!

**Comment by kitsuneDEV** [comment]
I'v tried it on the one you send and one for a TV show that contained musical note symbols. Those finally work now after scrubbing.

**Comment by duramato** [comment]
I'm still testing it, but I gave it a try on a show I have and it had missing characters with the force UTF-8 enabled:

```
2018-02-15 02:12:20,848 (6febf460) :  DEBUG (__init__:217) - :: ENCODING :: ISO-8859-1 ::
2018-02-15 02:12:26,056 (6febf460) :  DEBUG (__init__:215) - :: IGNORING BLOCK BECAUSE OF {LEGENDEI.COM</B>} FILTERED BY {.COM} ::
2018-02-15 02:12:26,086 (6febf460) :  INFO (__init__:273) - :: SCRUBBED :: /MEDIA/EXTSSD/SHOWS/DC'S LEGENDS OF TOMORROW/SEASON 03/DCS.LEGENDS.OF.TOMORROW.S03E08.CRISIS.ON.EARTH-X.4.1080P.WEB.X264-STRIFE.PT-BR.SRT ::
```

Small sample:
```
1
00:00:01,282 --> 00:00:03,872
Traduzido por legends
Airtonsub | may | rih | ranger

2
00:00:03,873 --> 00:00:06,121
Traduzido por legends
Prytti | nanda | avila

3
00:00:07,717 --> 00:00:10,403
Crise na terra-x
Parte 4

4
00:00:14,000 --> 00:00:16,297
O coroa est� ferido.
Temos que tir�-lo daqui!

5
00:00:21,710 --> 00:00:23,046
Cad� o barry quando
Precisamos?

6
00:00:23,967 --> 00:00:25,560
Tentando resolver
Umas coisas.
```

Do you want me to send you this subtitle file aswell?

**Comment by kitsuneDEV** [comment]
Do you see the missing character symbol "�" too when watching the movie?

Your other subtitle file turned into this for me: https://pastebin.com/VWtMX8tB
(keep in mind if have my own preferences) 
No 'question marks'

If you also see them when watching on Plex, please send me that sub file too and I'll have another look.

**Comment by duramato** [comment]
Ah no they don't, I was viewing the subtitles with `nano`, what got me on _alert_ was that I could read the char fine before being scrubbed. Sorry about that.

Mmm but now that I see, I have the force utf setting enabled,but:
```
$ file -bi /media/extssd/Shows/DC\'s\ Legends\ of\ Tomorrow/Season\ 03/DCs.Legends.of.Tomorrow.S03E08.Crisis.on.Earth-X.4.1080p.WEB.x264-strife.pt-BR.srt
text/plain; charset=iso-8859-1

```
And I've added costum words but they don't get used, it's as if it isn't using my configured settings, I even tried restarting Plex.

**Comment by kitsuneDEV** [comment]
I see, some editors wont handle it well and show that symbol.

You mean words in the filter input field? (From your other issue, or even plain-text ones?)

**Comment by duramato** [comment]
Even plain text ones, let me see if I can explain:
So I configured "sincronizadas" in the settings and saved them

![image](https://pls.gtfo.accountant/FvhDu.png) 

Then I forced a show  updated on a show I know had subtitled with that and 

```
2018-02-15 21:46:27,443 (7073f460) :  DEBUG (__init__:251) - :: ENCODING :: ISO-8859-1 ::
2018-02-15 21:46:32,842 (7073f460) :  INFO (__init__:273) - :: SCRUBBED :: /MEDIA/EXTSSD/SHOWS/TRUE DETECTIVE/SEASON 01/TRUE.DETECTIVE.S01E07.AFTER.YOUVE.GONE.1080P.BLURAY.X264-CHD.PT.SRT ::
```

But taking a look at the subtitle file:

```
577
00:50:37,694 --> 00:50:41,740
Daqui a 1 km, vire à esquerda
E apanhe a pr 1435.

578
00:50:43,325 --> 00:50:48,451
Depois de passar 10 km de campos,
Chega à 49, antes de crowley.

579
00:50:49,122 --> 00:50:51,875
- conhece bem estes caminhos.
- claro, chefe.

580
00:50:52,000 --> 00:50:54,208
Conheço toda a costa.
A minha família...

581
00:50:54,378 --> 00:50:55,789
Obrigado.

582
00:51:15,440 --> 00:51:18,561
A minha família está aqui
Há muito tempo.

583
00:52:27,982 --> 00:52:30,982
Ripadas e sincronizadas por:
Pt-subs rips
```
It's still there.

Also on a side note can SubsClear be more verbose on what it cleans? From what i've seen it only mentions when it's filtered by a tld

**Comment by kitsuneDEV** [comment]
Hm, that's very odd. I'll add a toggle in preferences for more verbose logging.
I'm still working on allowing accents into the filter lists. Will look into this at the end too. And run tests with different language files instead of the English I personally only need.

**Comment by kitsuneDEV** [comment]
I feel like it works now with "latin" accents. I can not get it to work with characters from say Russian or Arabic even after encoding those..
This might really be a thing other people need, but I'm kinda stuck on that.

v0.96 is available for testing.

**Comment by kitsuneDEV** [comment]
Should be solved in v0.97
